src/transformers/pipelines/token_classification.py

```diff
@@ -107,9 +108,10 @@ def __init__(
         self._args_parser = args_parser
         self.ignore_labels = ignore_labels
         self.grouped_entities = grouped_entities
+        self.subword_label_re_alignment = subword_label_re_alignment
         self.ignore_subwords = ignore_subwords
 
-        if self.ignore_subwords and not self.tokenizer.is_fast:
+        if (self.ignore_subwords or self.subword_label_re_alignment) and not self.tokenizer.is_fast:
```

Comment:
In terms of style in the rest of the lib, we usually have explicit bools in the test and never rely on truthy values. It would be great to store whether we have set a `subword_label_re_alignment` strategy in a boolean flag used for this test and the other ones like it, and `self.subword_label_re_alignment` have the actual strategy value.